Xwt.XamMac/Xwt.Mac/PanedBackend.cs

```diff
@@ -118,25 +120,83 @@ public override void ReplaceChild (NSView oldChild, NSView newChild)
 				Widget.AddSubview (view2);
 		}
 
-		public double Position {
-			get {
-				return 0;
+		private double position;
+		public double Position
+		{
+			get
+			{
+				return position;
 			}
-			set {
+			set
+			{
+				position = value;
+				this.DidResizeSubviews();
 			}
 		}
 		#endregion
 	}
-	
-	class CustomSplitView: NSSplitView, IViewObject
+
+	class CustomSplitView : NSSplitView, IViewObject
 	{
-		public NSView View {
-			get {
+		readonly double initDividerPosition = 220;
+		bool needsDividerSet;
+
+		public NSView View
+		{
+			get
+			{
 				return this;
 			}
 		}
 
-		public ViewBackend Backend { get; set; }
+		private ViewBackend backend;
+		public ViewBackend Backend
+		{
+			get
+			{
+				return backend;
+			}
+			set
+			{
+				if (value != null)
+				{
+					foreach (var view in ArrangedSubviews)
```

Comment:
Ugghh, how is the backend going to add back the arrangedsubviews? 

Should this just temporarily detach them?

Reply:
Thanks Thezok. It should be removed.